docs/express-vs-core.qmd

```diff
@@ -0,0 +1,262 @@
+---
+title: Express vs. Core
+editor:
+  markdown:
+    wrap: sentence
+---
+
+Shiny is one framework with two different syntax options: _Shiny Express_ and _Shiny Core_.
+Up until now, these docs have focused on Shiny Express.
+In this section, we'll dig into the differences between the two, and why you might choose one or the other.
+
+Don't worry, though.
+There is an enormous amount of overlap between the two, and the vast majority of apps will be well served by either Shiny Express or Shiny Core.
+
+And if you do decide to switch, the process is relatively straightforward---especially in the direction of Express to Core, as the latter supports a superset of the former's capabilities. We go into more detail on this in the [transitioning](express-to-core.qmd) article.
+
+## Background
+
+Shiny for Python was unveiled in 2022 with a single syntax option, which we now call Shiny Core.
+It drew inspiration from the [Shiny framework for R](https://shiny.posit.co/), which has been around for over a decade and is used by hundreds of thousands of data scientists around the world.
+Our goal was to make Shiny Core feel Pythonic, as opposed to a literal port of Shiny for R, but carry over the same core principles and tradeoffs.
+
+In January 2024, we introduced Shiny Express as a second syntax option.
+Express is built on top of Core, and is designed to be extremely easy to write, while preserving most of the power and flexibility of Shiny.
+
+## Differences between Express and Core
+
+The major differences between Shiny Express and Core are the following:
+
+- Slightly different import statements
+- Different organization of UI and server code
+- Implicit vs. explicit placement of outputs
+- Different syntax for UI containers
+
+Let's examine each of these in more detail.
+
+### Import statements
+
+A Shiny Core app file usually contains an import statement like:
+
+```python
+from shiny import App, reactive, render, ui
+```
+
+In Shiny Express, you'll instead see:
+
+```python
+from shiny import reactive
+from shiny.express import input, render, ui
+```
+
+Note that both import `ui` and `render`, but from different places.
+While Express's `shiny.express.ui` has almost all of the same UI functions as Core's `shiny.ui`, their function signatures often differ slightly, to reflect Express's different usage patterns.
+And the `render` functions---well actually, they are identical right now, but we're planning to add some Express-specific features to the `shiny.express.render` versions in the near future.
```

Comment:
I would cut this line as it'll become out of date, and the user doesn't really care that they're implemented the same way. Many of the `express.ui` functions are also re-exports.

Reply:
I planned to change this line when we change that functionality.